Assets/Scripts/Models/Area/WorldGenerator.cs

```diff
@@ -47,88 +40,66 @@ public static WorldGenerator Instance
-                int worldY = (height / 2) + startAreaCenterY - y;
-
-                Tile tile = world.GetTileAt(worldX, worldY, 0);
-
-                if (startAreaFurnitures[x, y] != null && startAreaFurnitures[x, y] != string.Empty)
+                float scaleZ = Mathf.Lerp(1f, .5f, Mathf.Abs((depth / 2f) - z) / depth);
+                for (int x = 0; x < width; x++)
                 {
-                    world.FurnitureManager.PlaceFurniture(startAreaFurnitures[x, y], tile, true);
-                }
-            }
-        }
-
-        for (int z = 0; z < depth; z++)
-        {
-            float scaleZ = Mathf.Lerp(1f, .5f, Mathf.Abs((depth / 2f) - z) / depth);
-            for (int x = 0; x < width; x++)
-            {
-                for (int y = 0; y < height; y++)
-                {
-                    float noiseValue = Mathf.PerlinNoise(
-                        (x + offsetX) / (width * asteroidInfo.NoiseScale * scaleZ),
-                        (y + offsetY) / (height * asteroidInfo.NoiseScale * scaleZ));
-                    if (noiseValue >= asteroidInfo.NoiseThreshhold && !IsStartArea(x, y, world))
+                    for (int y = 0; y < height; y++)
                     {
+                        float noiseValue = Mathf.PerlinNoise(
+                                           (x + offsetX) / (width * asteroidInfo.NoiseScale * scaleZ),
+                                           (y + offsetY) / (height * asteroidInfo.NoiseScale * scaleZ));
+                        
                         Tile tile = world.GetTileAt(x, y, z);
+                        if (noiseValue >= asteroidInfo.NoiseThreshhold && tile.Room != null && tile.Room.ID == 0)
+                        {
+                            tile.SetTileType(asteroidFloorType);
 
-                        tile.SetTileType(asteroidFloorType);
-
-                        world.FurnitureManager.PlaceFurniture("astro_wall", tile, false);
+                            world.FurnitureManager.PlaceFurniture("astro_wall", tile, false);
 
-                        if (Random.value <= asteroidInfo.ResourceChance && tile.Furniture.Name == "Rock Wall")
-                        {
-                            if (asteroidInfo.Resources.Count > 0)
+                            if (Random.value <= asteroidInfo.ResourceChance && tile.Furniture.Name == "Rock Wall")
                             {
-                                int currentweight = 0;
-                                int randomweight = Random.Range(0, sumOfAllWeightedChances);
-
-                                for (int i = 0; i < asteroidInfo.Resources.Count; i++)
+                                if (asteroidInfo.Resources.Count > 0)
                                 {
-                                    Resource inv = asteroidInfo.Resources[i];
+                                    int currentweight = 0;
+                                    int randomweight = Random.Range(0, sumOfAllWeightedChances);
```

Comment:
Didn't even notice that, leftovers from the previous version that got shifted around.